frontend/src/components/Profile/Profile.jsx

```diff
@@ -1,101 +1,377 @@
                         </div>
                     </div>
                 </div>
             </div>
-
-            <div className={styles["skills"]}>
-                <div className={styles["skills-detail"]}>
-                    <div className={styles["skills-header"]}>
-                        <h2> Skills </h2>
+  
+            <div className={styles["edit-profile-skill"]}>
+                <form>
+                    <div className={styles["skill-cards-row-container"]}>
+                        {cards.map((card) =>
+                            <Card 
+                                key={card.id}
+                                {...card}
+                                isEditing={editingCardId === card.id}
+                                onEditCard={() => handleEditCard(card.id)}
+                                onSavCarde={(editedSkillName, editedDescription) =>
+                                    handleSaveCard(card.id, editedSkillName, editedDescription)
+                                }
+                            />
+                        )}
                     </div>
+                </form>
+            </div>
+        </div>
+    );
+  };
 
-                    <div className={styles["mini-skills-info"]}>
-                        <div className={styles["skill-name"]}>
-                            <p> Skill Name </p>
-                        </div>
+const ProfileEdit = ({ onSaveProfile }) => {
+    // name
+    const [name, setName] = useState('');
 
-                        <div className={styles["years-of-experience"]}>
-                            <p> # years </p>
-                        </div>
-                    </div>
+    const handleNameChange = (e) => {
+        setName(e.target.value);
+    };
+
+    // location
+    const [location, setLocation] = useState('');
+
+    const handleLocationChange = (e) => {
+        setLocation(e.target.value);
+    };
+
+    // default profile pic
+    const avatarUrl = useRef(
+        "https://images2.alphacoders.com/130/1305995.png"
+    );
+
+    const updateAvatar = (imgSrc) => {
+        avatarUrl.current = imgSrc
+    }
+
+    const [modalOpen, setModalOpen] = useState(false)
+
+    // bio
+    const [bio, setBio] = useState('');
+    const MAX_TEXT_LENGTH = 500;
+    const handleBioChange = (e) => {
+        const value = e.target.value;
+        if (value.length <= MAX_TEXT_LENGTH) {
+            setBio(value);
+        };
+        setBio(e.target.value);
+    };
+
+    // rate
+    const [rate, setRate] = useState('');
+
+    const handleRateChange = (e) => {
+        setRate(e.target.value);
+    };
+
+    // tags
+    const [tag, setTag] = useState([]);
+
+    const handleTagChange = (e) => {
+        const trimmedValue = e.target.value.trim();
+
+        if (e.key === "Enter") {
+            e.preventDefault();
+
+            if (trimmedValue !== "") {
```

Comment:
What if the user adds more than one of the same tags? Do we want to add duplicate tags?

```suggestion
            if (trimmedValue !== "" && !tags.includes(trimmedValue)) {
```